jdate-class.js

```diff
@@ -133,9 +133,9 @@ JDate.prototype = {
 		var persian = this._persianDate();
 		persian[which] = value;
 		var new_date = jd_to_gregorian(persian_to_jd_fixed(persian[0], persian[1], persian[2]));
-		this._d.setFullYear(new_date[0]);
-		this._d.setMonth(new_date[1] - 1);
 		this._d.setDate(new_date[2]);
+		this._d.setMonth(new_date[1] - 1);
+		this._d.setFullYear(new_date[0]);
```

Comment:
because i have problem with this code and changing the ordering solves it :
~~~js
d = new JDate('1394-01-11');
console.log(d.toLocaleString());
d.setDate(12);
console.log(d.toLocaleString());
//it should be 1394/01/12 04:30:00 not *1394/02/11*
~~~
it seems it whould be better if we have a compeletly new object and convert the date as a whole , also we have such manner that you mentioned in original javaScripts Date too :
~~~js
d = new Date('2015-03-31');
console.log(d.toLocaleString());
// 3/31/2015, 4:30:00 AM
d.setMonth(3);
console.log(d.toLocaleString());
// 5/1/2015, 4:30:00 AM
~~~